temporalcli/commands_test.go

```diff
@@ -303,6 +308,15 @@ func StartDevServer(t *testing.T, options DevServerOptions) *DevServer {
 		}
 	}()
 
+	// Dial workflowservice client
+	d.GRPCConn, err = grpc.Dial(d.Options.ClientOptions.HostPort, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	require.NoError(t, err)
+	defer func() {
+		if !success {
+			d.GRPCConn.Close()
+		}
+	}()
```

Comment:
Can we remove this?

Reply:
Yes we can

Reply:
Note: I'm also going to change these to use `t.Cleanup()` in a followup PR so we don't need to track success